src/cpp/src/image_generation/stable_diffusion_3_pipeline.hpp

```diff
@@ -131,18 +131,25 @@ class StableDiffusion3Pipeline : public DiffusionPipeline {
 
         set_scheduler(Scheduler::from_config(root_dir / "scheduler/scheduler_config.json"));
 
+        // Temporary fix for GPU
+        ov::AnyMap updated_properties = properties;
+        if (device.find("GPU") != std::string::npos &&
+            updated_properties.find("INFERENCE_PRECISION_HINT") == updated_properties.end()) {
+            updated_properties["INFERENCE_PRECISION_HINT"] = ov::element::f32;
+        }
```

Comment:
let's introduce a function to get updated properties and re-use it in `compile` method as well for SD3 / SDXL